R/zchunk_L242.Bld_Inc_Elas_scenarios.R

```diff
@@ -73,6 +78,20 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios <- function(command, ...) {
           add_precursors("common/GCAM_region_names", "socioeconomics/A42.inc_elas",
                          "L102.pcgdp_thous90USD_Scen_R_Y") })
 
+    # L242.IncomeElasticity_bld_GCAM3: building sector income elasticity for GCAM 3.0 socioeconomics
+    # For the GCAM 3.0 scenario, calculate the per-capita GDP
+    L242.IncomeElasticity_bld_GCAM3 <- L102.gdp_mil90usd_GCAM3_R_Y %>%
+      left_join_error_no_match(L101.Pop_thous_GCAM3_R_Y, by = c("GCAM_region_ID", "year")) %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      transmute(region, year, pcgdp_90thousUSD = value.x / value.y) %>%
+      mutate(income.elasticity = approx(x = A42.inc_elas$pcgdp_90thousUSD, y = A42.inc_elas$inc_elas,
```

Comment:
We are basing the interpolation on an assumption file. To use approx_fun, I think I would need to bind the rows of the assumption files, probably after doing a `repeat_add_columns` for all regions, then `approx_fun`, then remove the assumption file data? Seems messy, but let me know if you have a cleaner method. Maybe adding all unique `pcgdp_90thousUSD` to the assumption file then doing approx_fun there would work.

This is also done in the file previously and in L252